stabilizer/src/main/java/com/hazelcast/stabilizer/tests/map/PredicatesTest.java

```diff
@@ -0,0 +1,438 @@
+package com.hazelcast.stabilizer.tests.map;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.*;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.SqlPredicate;
+import com.hazelcast.stabilizer.probes.probes.IntervalProbe;
+import com.hazelcast.stabilizer.tests.TestContext;
+import com.hazelcast.stabilizer.tests.TestRunner;
+import com.hazelcast.stabilizer.tests.annotations.*;
+import com.hazelcast.stabilizer.tests.map.helpers.StringUtils;
+import com.hazelcast.stabilizer.tests.utils.TestUtils;
+import com.hazelcast.stabilizer.tests.utils.ThreadSpawner;
+import com.hazelcast.stabilizer.worker.Metronome;
+import com.hazelcast.stabilizer.worker.OperationSelector;
+import com.hazelcast.stabilizer.worker.SimpleMetronome;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class PredicatesTest {
+
+    private static enum PredicateType {
+        NAME,
+        AGE,
+        ACTIVE,
+        SALARY,
+    }
+
+    private final static ILogger log = Logger.getLogger(PredicatesTest.class);
+    private final AtomicLong operations = new AtomicLong();
+    // properties
+    public int threadCount = 40;
+    public int keyLength = 10;
+    // number of keys per member
+    public int keyCount = 10000;
+    public int logFrequency = 10000;
+    public int performanceUpdateFrequency = 1;
+
+    //default
+    /**
+    DATA_SERIALIZABLE = 1
+    JAVA_SERIALIZABLE = 2
+    IDENTIFIED_DATA_SERIALIZABLE = 3
+    PORTABLE = 4
+    */
+    public int serializationStrategyWithNumber = 3;
+    public boolean customPredicate = false;
+    public String sqlQuery = "age = 30 AND active = true";
+    public double nameProb= 0.0;
```

Comment:
I don't see the added value between switching between predicates if all the predicates are practically the same from a complexity point of view. Perhaps I'm missing something; otherwise this should go.